File: tests/fixtures/multilang/sample.py
"""A sample Python module for parser testing"""

import os
from typing import Optional


class Config:
    """Configuration holder"""

    def __init__(self, name: str, value: int):
        self.name = name
        self.value = value

    def validate(self) -> bool:
        """Check if config is valid"""
        return len(self.name) > 0 and self.value > 0


class AdvancedConfig(Config):
    """Extended configuration with extra features"""

    def __init__(self, name: str, value: int, debug: bool = False):
        super().__init__(name, value)
        self.debug = debug

    def validate(self) -> bool:
        """Override validation with additional checks"""
        return super().validate() and isinstance(self.debug, bool)


def process_config(config: Config) -> Optional[str]:
    """Process a config and return result"""
    if config.validate():
        return f"Processed: {config.name}"
    return None


MAX_RETRIES = 3

File: tests/fixtures/multilang/sample.rb
# A sample Ruby module for parser testing

module Sample
  class Config
    attr_reader :name, :value

    def initialize(name, value)
      @name = name
      @value = value
    end

    def validate
      !name.nil? && !name.empty? && value > 0
    end
  end

  class AdvancedConfig < Config
    def initialize(name, value, debug: false)
      super(name, value)
      @debug = debug
    end

    def validate
      super && @debug.is_a?(TrueClass) || @debug.is_a?(FalseClass)
    end
  end

  def self.process_config(config)
    if config.validate
      "Processed: #{config.name}"
    end
  end
end
